runtime-modules/forum/src/lib.rs

```diff
@@ -1316,16 +1316,31 @@ impl<T: Trait> Module<T> {
         ensure!(category_id != new_category_id, ERROR_THREAD_MOVE_INVALID,);
 
         let info = Self::ensure_can_moderate_thread(origin, moderator_id, category_id, thread_id);
-        if info.is_err() {
-            return Err(ERROR_MODERATOR_MODERATE_ORIGIN_CATEGORY);
+
+        match info {
```

Comment:
This is indeed safe, but its a bit hard to follow, try this

```Rust
let (account_id, thread) = Self::ensure_can_moderate_thread(..)
.map_err(ERROR_MODERATOR_MODERATE_ORIGIN_CATEGORY)?;

Self::ensure_can_moderate_category(..)
.map_err(ERROR_MODERATOR_MODERATE_DESTINATION_CATEGORY)?;

Ok(..)
```

Reply:
Much better, thx.